exo-go/src/app_deployer/index.go

```diff
@@ -12,27 +14,51 @@ import (
 type AppDeployer struct {
 	AppConfig      types.AppConfig
 	ServiceConfigs map[string]types.ServiceConfig
-	Logger         *logger.Logger
 	AppDir         string
 	HomeDir        string
+	Logger         *logger.Logger
 }
 
 // NewAppDeployer is the constructor for the appDeployer class
-func NewAppDeployer(appConfig types.AppConfig, serviceConfigs map[string]types.ServiceConfig, appDir, homeDir string) *AppDeployer {
+func NewAppDeployer(appConfig types.AppConfig, serviceConfigs map[string]types.ServiceConfig, appDir, homeDir string, logger *logger.Logger) *AppDeployer {
 	return &AppDeployer{
 		AppConfig:      appConfig,
 		ServiceConfigs: serviceConfigs,
 		AppDir:         appDir,
 		HomeDir:        homeDir,
+		Logger:         logger,
 	}
 }
 
 // Start starts the deployment process
 func (d *AppDeployer) Start() error {
 	terraformDir := getTerraformDir(d.AppDir)
-	return terraformFileHelpers.GenerateTerraformFile(d.AppConfig, d.ServiceConfigs, d.AppDir, d.HomeDir, terraformDir)
+
+	err := terraformFileHelpers.GenerateTerraformFile(d.AppConfig, d.ServiceConfigs, d.AppDir, d.HomeDir, terraformDir)
+	if err != nil {
+		return err
+	}
+
+	err = terraformCommandHelpers.TerraformInit(terraformDir, d.write)
+	if err != nil {
+		return err
+	}
+
+	secretsFile := getSecretsFile(d.AppDir)
+	return terraformCommandHelpers.TerraformPlan(terraformDir, secretsFile, d.write)
+}
+
+func (d *AppDeployer) write(text string) {
+	err := d.Logger.Log("exo-deploy", text, true)
+	if err != nil {
+		fmt.Printf("Error logging exo-deploy output: %v\n", err)
+	}
 }
 
 func getTerraformDir(appDir string) string {
```

Comment:
Sorry I missed this but I expected this to be function on the `AppDeployer` struct and thus appDir wouldn't need to be passed in. `getSecretsFile` should also use this function since the secrets is in the terraform dir